tst.c

```diff
@@ -4,6 +4,26 @@
 #define WRDMAX 128
 #define STKMAX (WRDMAX * 2)
 
+/* This macro is used to replace some repeating pattern for rotating and
+ * deleting the node on tenary search tree. It will append kid of victim on
+ * suitable position, then free victim itself. Note that 'kid' should be
+ * victim->lokid or victim->hikid
+ */
+#define del_node(parent, victim, root, kid) \
+    do {                                    \
+        if (!parent)                        \
+            *root = kid;                    \
+        else if (victim == parent->lokid)   \
```

Comment:
What if `parent` is NULL and `parent->lokid` is referred?

Reply:
If `parent` is NULL, it means that `victim` is the current root. So the statement `*root = kid` will execute which will change the current root to the kid of victim. Things like `parent->lokid` won't be executed in this scenario. I'm not if the answer is what you ask for?
 
The purpose of this macro is to reduce duplicated code, I'm not sure if this is a good idea to do it like this. If this is a implementation, I can expend the macro back to original code.

Reply:
I would expect the sequence like:
```c
if (!parent) {
    *root = kid;
} else {
    if (victim == parent->lokid)
        parent->lokid = kid;
    else if (victim == parent->hikid)
        parent->hikid = kid;
    else
        parent->eqkid = kid;
}
```
to improve readability.

Reply:
Thank for the opinion. I improve it now.